notify/msteamsv2/msteamsv2.go

```diff
@@ -0,0 +1,193 @@
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package msteamsv2
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"strings"
+
+	"github.com/go-kit/log"
+	"github.com/go-kit/log/level"
+	commoncfg "github.com/prometheus/common/config"
+	"github.com/prometheus/common/model"
+
+	"github.com/prometheus/alertmanager/config"
+	"github.com/prometheus/alertmanager/notify"
+	"github.com/prometheus/alertmanager/template"
+	"github.com/prometheus/alertmanager/types"
+)
+
+const (
+	colorRed   = "Attention"
+	colorGreen = "Good"
+	colorGrey  = "Warning"
+)
+
+type Notifier struct {
+	conf         *config.MSTeamsV2Config
+	tmpl         *template.Template
+	logger       log.Logger
+	client       *http.Client
+	retrier      *notify.Retrier
+	webhookURL   *config.SecretURL
+	postJSONFunc func(ctx context.Context, client *http.Client, url string, body io.Reader) (*http.Response, error)
+}
+
+// https://learn.microsoft.com/en-us/connectors/teams/?tabs=text1#adaptivecarditemschema
+type Content struct {
+	Schema  string `json:"$schema"`
+	Type    string `json:"type"`
+	Version string `json:"version"`
+	Body    []Body `json:"body"`
+}
+
+type Body struct {
+	Type   string `json:"type"`
+	Text   string `json:"text"`
+	Weight string `json:"weigth,omitempty"`
+	Size   string `json:"size,omitempty"`
+	Wrap   bool   `json:"wrap,omitempty"`
+	Style  string `json:"style,omitempty"`
+	Color  string `json:"color,omitempty"`
+}
+
+type Attachment struct {
+	ContentType string  `json:"contentType"`
+	ContentURL  *string `json:"contentUrl"` // Use a pointer to handle null values
+	Content     Content `json:"content"`
+}
+
+type teamsMessage struct {
+	Type        string       `json:"type"`
+	Attachments []Attachment `json:"attachments"`
+}
+
+// New returns a new notifier that uses the Microsoft Teams Power Platform connector.
+func New(c *config.MSTeamsV2Config, t *template.Template, l log.Logger, httpOpts ...commoncfg.HTTPClientOption) (*Notifier, error) {
+	client, err := commoncfg.NewClientFromConfig(*c.HTTPConfig, "msteamsv2", httpOpts...)
+	if err != nil {
+		return nil, err
+	}
+
+	n := &Notifier{
+		conf:         c,
+		tmpl:         t,
+		logger:       l,
+		client:       client,
+		retrier:      &notify.Retrier{},
+		webhookURL:   c.WebhookURL,
+		postJSONFunc: notify.PostJSON,
+	}
+
+	return n, nil
+}
+
+func (n *Notifier) Notify(ctx context.Context, as ...*types.Alert) (bool, error) {
+	key, err := notify.ExtractGroupKey(ctx)
+	if err != nil {
+		return false, err
+	}
+
+	level.Debug(n.logger).Log("incident", key)
```

Comment:
What is incident referring to? Is this left over from testing?

Reply:
Is not great - but it's common in other receivers: https://github.com/search?q=repo%3Aprometheus%2Falertmanager%20%22incident%22&type=code

I left it as is because at least it aligns with others.